cmd/zed/agents/zfs_mod.c

```diff
@@ -223,8 +243,15 @@ zfs_process_add(zpool_handle_t *zhp, nvlist_t *vdev, boolean_t labeled)
 		}
 	}
 
+	if (!zpool_get_prop_int(zhp, ZPOOL_PROP_AUTOREPLACE, NULL)) {
```

Comment:
Auto-online doesn't require opting in.  So this check breaks auto-online for the normal/default case.